### PR TITLE
Fixed a bug where key did not work on redhat due to incorrect pkg name

### PR DESCRIPTION
**Pull request by fhaynes**
Replaced the hardcoded package name with the package variable.

### CODE REVIEW

File: manifests/key.pp [hunk 1]
```diff
@@ -15,7 +15,7 @@
     command     => "/usr/sbin/dnssec-keygen -a HMAC-MD5 -r /dev/urandom -b 128 -n USER ${name}",
     cwd         => "${cfg_dir}/bind.keys.d",
     require     => [
-      Package['dnssec-tools','bind9'],
+      Package['dnssec-tools',$package],
```
Comment (solarkennedy) [inline manifests/key.pp]: I don't understand, where does this variable come from? I don't see it in the namespace?

Comment (fhaynes) [inline manifests/key.pp]: It is from dns::server::params I believe.

Comment (solarkennedy) [inline manifests/key.pp]: So.. shouldn't it be dns::server::package or something?

Alternatively I would argue that this should just require 'dnssec-tools' and let the other classes require their own packages?

Comment (fhaynes) [inline manifests/key.pp]: My apologies, when I was preparing a clean branch for the pull request I neglected to include the $package = $dns::server::params::package line in the file, so you're right, $package would have been undef. I pushed a new commit that fully qualifies it, opting to just fully qualify it in the Package require.

Regarding just requiring dnssec-tools, that is probably a better choice. I was just doing the minimum necessary to fix the bug and get it working on RedHat while attempting to preserve the function of the original code. If you'd rather go that route, I'll push a commit that removes the requirement for the bind package entirely.

Comment (solarkennedy) [inline manifests/key.pp]: Yea, the dnssec-tools is the actual requirement, as attached to the exec. It doesn't actually need to depend on bind.

**Pull request merged**